File: components/lfric-xios/integration-test/lfric_xios_context_test.py
#!/usr/bin/env python3
##############################################################################
# (C) Crown copyright 2024 Met Office. All rights reserved.
# The file LICENCE, distributed with this code, contains details of the terms
# under which the code may be used.
##############################################################################
"""
Simple test which initialises a minimal `lfric_xios_context_type` object and
then destroys it. This will also create an attached XIOS context.
"""

from testframework import TestEngine, TestFailed
from xiostest import LFRicXiosTest
import sys

###############################################################################
class LfricXiosContextTest(LFRicXiosTest):
    """
    Tests the lfric_xios_context_type by creating and destroying it
    """

    def __init__(self):
        super().__init__(command=[sys.argv[1], "context.nml"], processes=1)
        self.gen_config( "context.nml", "context.nml", {} )

    def test(self, returncode: int, out: str, err: str):
        """
        Test the output of the context test
        """

        if returncode != 0:
            raise TestFailed(
                f"Unexpected failure of test executable: {returncode}\n"
                "stderr:\n"
                f"{err}"
            )

        for xios_out in self.xios_out:
            if not xios_out.exists():
                raise TestFailed("XIOS context log output not found")
            if (
                "-> info : Client side context is finalized\n"
                not in xios_out.contents
            ):
                raise TestFailed("XIOS context not finalised")

        for xios_err in self.xios_err:
            if not xios_err.exists():
                raise TestFailed("XIOS context log err not found")

        return "XIOS context initialised and destroyed successfully"


##############################################################################
if __name__ == "__main__":
    TestEngine.run(LfricXiosContextTest())

File: components/lfric-xios/integration-test/lfric_xios_context_test.f90
!-----------------------------------------------------------------------------
! (C) Crown copyright 2024-2025 Met Office. All rights reserved.
! The file LICENCE, distributed with this code, contains details of the terms
! under which the code may be used.
!-----------------------------------------------------------------------------

! Simple test which initialises and destroys an XIOS context
!
program lfric_xios_context_test

  use field_mod,              only: field_type
  use lfric_xios_context_mod, only: lfric_xios_context_type
  use lfric_xios_driver_mod,  only: lfric_xios_initialise, lfric_xios_finalise
  use log_mod,                only: log_event, log_level_info
  use test_db_mod,            only: test_db_type

  use local_mesh_mod, only: local_mesh_type
  use mesh_mod, only: mesh_type

  implicit none

  type(test_db_type)                         :: test_db
  type(lfric_xios_context_type), allocatable :: io_context


  call test_db%initialise()
  call lfric_xios_initialise( "test", test_db%comm, .false. )

  ! =============================== Start test ================================

  allocate(io_context)
  call io_context%initialise( "test_io_context", 1, 10 )
  call io_context%initialise_xios_context( test_db%comm, &
                                           test_db%chi, test_db%panel_id, &
                                           test_db%clock, test_db%calendar, &
                                           test_db%config%base_mesh%geometry(), &
                                           test_db%config%base_mesh%topology(), &
                                           test_db%config%finite_element%coord_system(), &
                                           test_db%config%planet%scaled_radius() )

  deallocate(io_context)

  ! ============================== Finish test =================================

  call lfric_xios_finalise()
  call test_db%finalise()

end program lfric_xios_context_test
